src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -186,84 +216,127 @@ public void Dispose()
         /// or whitespace separated list, with each entry in the following format:
         /// ([&lt;scheme&gt;=][&lt;scheme&gt;"://"]&lt;server&gt;[":"&lt;port&gt;])
         /// </summary>
-        private static void ParseProxyConfig(string value, out Uri insecureProxy, out Uri secureProxy)
+        private static void ParseProxyConfig(ReadOnlySpan<char> proxyString, out MultiProxy secureProxy, out MultiProxy insecureProxy)
         {
-            insecureProxy = null;
-            secureProxy = null;
-            if (string.IsNullOrEmpty(value))
-            {
-                return;
-            }
+            Uri[] secureUris = new Uri[4], insecureUris = new Uri[4];
+            int secureUriCount = 0, insecureUriCount = 0;
+            int iter = 0;
 
-            if (value.IndexOfAny(s_proxyDelimiters) == -1)
+            while (true)
```

Comment:
'while(true)' loops scare me. :)

Is it possible that the data which will come from a PAC file (and be tampered with or have errors in it) could cause the loop to run forever?  Do we have sufficient test cases around this?

Reply:
I don't see anything that would cause an infinite loop here, but additional tests are never a bad thing. I don't think we have any failure/malicious case tests for config parsing.

Reply:
Github shows as outdated, but only because this code was moved into MultiProxy.cs